exampleMultipartyVideo/src/main/java/com/twilio/examplemultipartyvideo/RoomActivity.java

```diff
@@ -0,0 +1,1010 @@
+        if (localVideoTrack != null) {
+            connectOptionsBuilder.videoTracks(Collections.singletonList(localVideoTrack));
+        }
+
+        /*
+         * Set the preferred audio and video codec for media.
+         */
+        connectOptionsBuilder.preferAudioCodecs(Collections.singletonList(audioCodec));
+        connectOptionsBuilder.preferVideoCodecs(Collections.singletonList(videoCodec));
+
+        /*
+         * Set the sender side encoding parameters.
+         */
+        connectOptionsBuilder.encodingParameters(encodingParameters);
+
+        room = Video.connect(this, connectOptionsBuilder.build(), roomListener());
+        setDisconnectAction();
+    }
+
+    /*
+     * The initial state when there is no active room.
+     */
+    private void intializeUI() {
+        connectActionFab.setImageDrawable(ContextCompat.getDrawable(this,
+                R.drawable.ic_video_call_white_24dp));
+        connectActionFab.show();
+        connectActionFab.setOnClickListener(connectActionClickListener());
+        localVideoActionFab.show();
+        localVideoActionFab.setOnClickListener(localVideoClickListener());
+        muteActionFab.show();
+        muteActionFab.setOnClickListener(muteClickListener());
+    }
+
+    /*
+     * The actions performed during disconnect.
+     */
+    private void setDisconnectAction() {
+        connectActionFab.setImageDrawable(ContextCompat.getDrawable(this,
+                R.drawable.ic_call_end_white_24px));
+        connectActionFab.show();
+        connectActionFab.setOnClickListener(disconnectClickListener());
+    }
+
+    /*
+     * Creates an connect UI dialog
+     */
+    private void showConnectDialog() {
+        EditText roomEditText = new EditText(this);
+        connectDialog = Dialog.createConnectDialog(roomEditText,
+                connectClickListener(roomEditText),
+                cancelConnectDialogClickListener(),
+                this);
+        connectDialog.show();
+    }
+
+    private void initializeVideoTextureViews() {
+        localVideoTextureView = findViewById(R.id.videoView1);
+        localVideoTextureView.setMirror(true);
+
+        videoViewMap.clear();
+        videoViewMap.put(localParticipant, localVideoTextureView);
+
+        availableVideoTextureViews.clear();
+
+        VideoTextureView videoTextureView4 = findViewById(R.id.videoView4);
+        videoTextureView4.setVisibility(GONE);
+        availableVideoTextureViews.push(videoTextureView4);
+
+        VideoTextureView videoTextureView3 = findViewById(R.id.videoView3);
+        videoTextureView3.setVisibility(GONE);
+        availableVideoTextureViews.push(videoTextureView3);
+
+        VideoTextureView videoTextureView2 = findViewById(R.id.videoView2);
+        videoTextureView2.setVisibility(GONE);
+        availableVideoTextureViews.push(videoTextureView2);
+    }
+
+    private VideoTextureView getAvailableVideoTextureView() {
+        if (availableVideoTextureViews.size() == 0) {
```

Comment:
Just throw an exception if this happens because this example does not support more than MAX_PARTICIPANTS